icu4c/source/test/cintltst/ulocaletst.c

```diff
@@ -22,48 +22,40 @@ enum {
     NAME,
     BASENAME,
 };
-static const char* const rawData[][9] = {
-    // language code
-    {   "en",   "fr",   "ca",   "el",   "no",   "it",   "xx",   "zh",
-         "x-klingon"  },
-    // script code
-    {   "",     "",     "",     "",     "",     "",     "",     "Hans",
-        "Latn"},
-    // country code
-    {   "US",   "FR",   "ES",   "GR",   "NO",   "",     "YY",   "CN",
-        "ZX"},
-    // variant code
-    {   "",     "",     "",     "",     "NY",   "",     "",   "",
-        ""},
-    // full name
-    {   "en_US",    "fr_FR",    "ca_ES",    "el_GR",    "no_NO_NY", "it",   "xx_YY",   "zh_Hans_CN",
-        "x-klingon_Latn_ZX.utf32be@special"},
-    // basename
-    {   "en_US",    "fr_FR",    "ca_ES",    "el_GR",    "no_NO_NY", "it",   "xx_YY",   "zh_Hans_CN",
-        "x-klingon_Latn_ZX.utf32be@special"},
+static const char* const rawData[][6] = {
+    { "en", "", "US", "", "en_US", "en_US"},
+    { "fr", "", "FR", "", "fr_FR", "fr_FR"},
+    { "ca", "", "ES", "", "ca_ES", "ca_ES"},
+    { "el", "", "GR", "", "el_GR", "el_GR"},
+    { "no", "", "NO", "NY", "no_NO_NY", "no_NO_NY"},
+    { "it", "", "", "", "it", "it"},
+    { "xx", "", "YY", "", "xx_YY", "xx_YY"},
+    { "zh", "Hans", "CN", "", "zh_Hans_CN", "zh_Hans_CN"},
+    { "zh", "Hans", "CN", "", "zh_Hans_CN", "zh_Hans_CN"},
+    { "x-klingon", "Latn", "ZX", "", "x-klingon_Latn_ZX.utf32be@special", "x-klingon_Latn_ZX.utf32be@special"},
 };
 
 static void TestBasicGetters() {
-    for (int32_t i = 0; i < UPRV_LENGTHOF(rawData[0]); i++)  {
-        ULocale* l = ulocale_openForLocaleID(rawData[NAME][i], -1);
-        assertEquals("ulocale_getLanguage()", rawData[LANGUAGE][i], ulocale_getLanguage(l));
-        assertEquals("ulocale_getScript()", rawData[SCRIPT][i], ulocale_getScript(l));
-        assertEquals("ulocale_getRegion()", rawData[REGION][i], ulocale_getRegion(l));
-        assertEquals("ulocale_getVariant()", rawData[VAR][i], ulocale_getVariant(l));
-        assertEquals("ulocale_getLocaleID()", rawData[NAME][i], ulocale_getLocaleID(l));
-        assertEquals("ulocale_getBaseName()", rawData[BASENAME][i], ulocale_getBaseName(l));
+    for (int32_t i = 0; i < UPRV_LENGTHOF(rawData); i++)  {
+        UErrorCode status = U_ZERO_ERROR;
+        ULocale* l = ulocale_openForLocaleID(rawData[i][NAME], -1, &status);
+        assertSuccess(WHERE "ulocale_openForLocaleID()", &status);
```

Comment:
`assertSuccess()` returns true if the error code passed `U_SUCCESS`, and you probably want to check that before doing all the other tests, since if `ulocale_openForLocaleID()` fails, `l` is bad.

That is...

```
if (assertSuccess(WHERE "ulocale_openForLocaleID()", &status) {
    assertEquals(WHERE "ulocale_getLanguage()", rawData[i][LANGUAGE], ulocale_getLanguage(l));
    assertEquals(WHERE "ulocale_getScript()", rawData[I][SCRIPT], ulocale_getScript(l));
    // etc...
    ulocale_close(l);
}
```